_transcripts/039-rust-1.54-1.55.md

```diff
@@ -0,0 +1,718 @@
+error all along— because it was really a bug, but we just haven't made it one
+yet. Or at a future edition boundary, this behavior will change. So we're giving
+you a heads-up now, in case you're thinking of adopting the future edition.
+
+And there are a lot of different things that are matched under this. So as an
+example, the array `into_iter` that we talked about, I think last episode, is a
+future-incompatibility lint, that you're relying here on this auto-ref behavior,
+and in the next edition that won't be there anymore. Or there's like— there's a
+lint for constants whose values are impossible. So for example, if you declare a
+constant whose value is like, one divided by zero, then like, it used to be that
+you could write Rust code that declared such a constant, and it would only be an
+error when you use the constant, and not when you declare it. But in the future,
+it's probably going to become a hard error to even declare one. And so this is
+an example of a future-compatibility lint, where you can do this today, but it's
+really a bug, and in the future, it's going to become a hard error. And so this
+is like, a way for you as a user to be like, I want to be aware of things that
+are coming down the pike. Please Cargo, tell me about them.
+
+__Ben__: Yeah, we should clarify too, that— so normally all of these
+incompatibilities are just warnings, and you should be seeing them by default in
+your own code. But what `cargo report`, I believe is doing, is that it is
+showing warnings on your dependencies. Normally, Cargo does not display warnings
+that are in your dependencies. So for example, this could be a problem if
+there's some kind of future incompatibility in code that you— a crate that
+you're using, but it's not printing the warning. So you have no idea if you're
+supposed to be upgrading, or if it will break under you, which actually— the
+fact that Cargo hasn't been doing this is one of the reasons why a lot of these
+incompatibility lints haven't actually progressed very far, and some of them
+have been open for a few years now, just because Cargo hasn't really been up the
+task of warning users that their code might break. And so hopefully, this
+represents a step towards making it so that we can actually move forward on
+these incompatibility lints that have been kind of languishing for years and
+years now.
+
+__Jon__: Yeah, and part of the hope too, I assume, is that because you get to
+see it in your dependencies, you can then go and try to fix them. Like you can
+submit either an issue or a PR to your sort of upstream dependencies and go,
+hey, I was told about this lint from Cargo, here's a thing that just fixes it so
+that your code won't break in the future.
+
+__Ben__: One more little thing that I noticed— not in the release notes, but
+secretly, is that 1.54 has turned on the `mutable_noalias` flag once again, by
+default, on certain LLVM versions. And so we've maybe spoken about this before
```

Comment:
I'm not certain whether this is "mutable noalias" or `mutable_noalias`.

(14:20)

Reply:
`mutable-noalias` technically (https://github.com/rust-lang/rust/pull/88325), but "mutable noalias" works well I think.